cli/src/trusted_operation.rs

```diff
@@ -287,63 +299,91 @@ fn send_direct_request<T: Decode + Debug>(
 	let (sender, receiver) = channel();
 	direct_api.watch(jsonrpc_call, sender);
 
-	debug!("waiting for rpc response");
+	// the first response of `submitAndWatch` is just the plain top hash
+	let top_hash = await_subscription_response(&receiver).map_err(|e| {
+		error!("Error getting subscription response: {:?}", e);
+		direct_api.close().unwrap();
+		e
+	})?;
+
+	debug!("subscribing to updates for top with hash: {top_hash:?}");
+
 	loop {
-		match receiver.recv() {
-			Ok(response) => {
-				debug!("received response");
```

Comment:
Simply remove the nesting, by replacing in `match` and `if let` with returning early with `?` the behavior was preserved.